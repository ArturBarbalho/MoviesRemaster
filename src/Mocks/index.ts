import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }