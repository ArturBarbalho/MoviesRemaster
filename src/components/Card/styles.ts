import styled from 'styled-components'
import media from 'styled-media-query'
export const Wrapper = styled.div`
`
export const Over = styled.div`
opacity:0;
:Hover{
    opacity:0.5;
    cursor:pointer;
}
position:absolute;
background-color:black;
width:350px;
height: 515px;
@media(max-width:450px){
        width:205px;
    } 
@media(max-width:400px){
        width:160px;
        height:233px;
    } 

`