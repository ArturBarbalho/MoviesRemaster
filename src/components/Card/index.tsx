import { Functions } from '../../getdata'
import { CardProps } from '../../typesProps'
import Image from '../Image'
import * as S from './styles'
type props = {props: CardProps}
const Card = ({props}:props) => (
  <S.Wrapper>
    <S.Over data-testid={'over'} onClick={()=>Functions.Details(props.title,props.id,props.media)} />
    <Image size='small' src={props.src} />
  </S.Wrapper>
)

export default Card

