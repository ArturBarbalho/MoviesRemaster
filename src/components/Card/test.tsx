import { render, screen } from '@testing-library/react' 
import Card from '.'
import { CardMock } from '../../Mocks'

describe('<Card />',  () => {
  it('should appear when the mouse is hover', () => {
    const { container } = render(<Card props={CardMock} />)

    expect(screen.getByTestId('over')).toHaveStyle(
      'position:absolute;'
    )  

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the image', () => {
    const { container } = render(<Card props={CardMock} />)

    expect(screen.getByRole('img')).toBeInTheDocument()


    expect(container.firstChild).toMatchSnapshot()
  })

})