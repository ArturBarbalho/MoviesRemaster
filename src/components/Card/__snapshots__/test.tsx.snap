// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should appear when the mouse is hover 1`] = `
<div
  class="styles__Wrapper-sc-1xbaftz-0 cSoiqW"
>
  <div
    class="styles__Over-sc-1xbaftz-1 fRvyVk"
    data-testid="over"
  />
  <img
    class="styles__Wrapper-sc-y6e3xk-0 fOWEhk"
    src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
  />
</div>
`;

exports[`<Card /> should render the image 1`] = `
<div
  class="styles__Wrapper-sc-1xbaftz-0 cSoiqW"
>
  <div
    class="styles__Over-sc-1xbaftz-1 fRvyVk"
    data-testid="over"
  />
  <img
    class="styles__Wrapper-sc-y6e3xk-0 fOWEhk"
    src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
  />
</div>
`;
